app/src/main/java/com/example/intandem/fragments/PostsFragment.java

```diff
@@ -207,18 +354,82 @@ public void done(List<Post> posts, ParseException e) {
                             }
                         }
                         allPosts.addAll(posts);
-                        adapter.notifyDataSetChanged();
+                        if (maxDistance == -1) {
+                            adapter.notifyDataSetChanged();
+                            return;
+                        }
+                        // FILTERING BY DISTANCE
+                        filteredDistancePosts.clear();
+
+                        Retrofit retrofit = new Retrofit.Builder()
+                                .baseUrl(BASE_URL)
+                                .addConverterFactory(GsonConverterFactory.create())
+                                .build();
+                        StringBuilder destinations = new StringBuilder();
+                        for (int i = 0; i < allPosts.size() - 1; i++) {
+                            CustomPlace customPlace = allPosts.get(i).getCustomPlace();
+                            try {
+                                customPlace.fetchIfNeeded();
+                                destinations.append("place_id:").append(customPlace.getGPlaceId()).append("|");
```

Comment:
A comment explaining what this is doing would help its readability.  Why do we need to create this custom string?